bin/make-test

```diff
@@ -1,10 +1,12 @@
-#! /bin/sh
+#!/usr/bin/env bash
+
+set -eu -o pipefail
 
 go_test_output="/tmp/go-test.out"
 
 go test -v github.com/trussworks/truss-aws-tools/... | tee "${go_test_output}"
 
-if [ -n "$CIRCLECI" ]; then
+if [[ -n ${CIRCLECI-} ]]; then
```

Comment:
What's with the dash in the `${CIRCLECI-}` here?

Reply:
Because I enabled `set -u` ${CIRCLECI} will error our as an unbound variable because it's unset outside of CircleCI. The empty dash will substitute a default value. In this case I'm substituting nothing but a clearer example is below where $SHELL is defined and outputs `/bin/zsh`, but $UNDEFINED is not and returns the substituted default value `somedefaultvalue`

```bash
echo ${SHELL-somedefaultvalue}
/bin/zsh
echo ${UNDEFINED-somedefaultvalue}
somedefaultvalue
```

I'll add some comments to make this more clear.

Reply:
Aha. I have not used that before. Okay, wfm.